.github/workflows/LINUX_BUILD_TEST.yml

```diff
@@ -0,0 +1,736 @@
+name: Linux
+
+on:
+  workflow_call:
+    inputs:
+      SIMENG-MODE:
+        required: true
+        type: string
+      RUNNER:
+        default: ubuntu-latest
+        required: true
+        type: string
+
+env:
+  ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION: true
+  LLVM-VERSION: 14
+  BENCHMARK_BRANCH: 'make-file-build-system' # The branch inside the benchmark repo that has the script to run all benchmarks.
+  PAT: ${{ secrets.SIMENGUOB_PAT }}
+
+
+jobs:
+  Build_and_Run:
+    runs-on: ${{ inputs.RUNNER }}
+    
+    strategy:
+      fail-fast: false
+      matrix:
+        
+        COMPILER: ['gcc-7', 'gcc-8', 'gcc-9', 'gcc-10'] # todo 'armclang'] # compiler names
+        OS: ['ubuntu:18.04','ubuntu:20.04', 'rockylinux:8', 'redhat/ubi8:latest', 'redhat/ubi9:latest', 'debian:10', 'debian:11'] # Docker images
```

Comment:
Depending on current runtime, we can probably get away with just redhat 9 and rocky 8 (or I'd personally prefer to see rocky 9 and redhat 8). RHEL and Rocky are very similar OS's (designed to be 1:1 compatible), so a bit redundant to have two of the "same", but worth testing both 8 and 9 as well as RHEL and Rocky in _some_ combination

Reply:
Can be changed at a later date if we run out of runners